hasura/graphql-engine
# #4736 Improvements to auth testing

### Description
One minor user-facing change. The second commit is just a refactor for ease of unit testing (and as a consequence, better code organization hopefully)

### Changelog


### Affected components



### Related Issues
#4736  ... it's not clear if this is sufficient to close that issue

### Solution and Design

Simple factoring out of tricky bits to make unit tests of core auth logic possible. This would also be the first step towards more clever solutions like e.g. having a JWT transformer/effect, but I don't think that gives us much.

### Steps to test and verify
- Please read through test spec carefully, make sure actual behavior is desired
- make note of change mentioned in changelog 
- look closely at big block moves (with poor diff; sorry about that), to make sure they are in fact trivial refactors

### Limitations, known bugs & workarounds



### Server checklist


#### Catalog upgrade

Does this PR change Hasura Catalog version?

#### Metadata


Does this PR add a new Metadata feature?
  - Does `run_sql` auto manages the new metadata through schema diffing?
  - Does `run_sql` auto manages the definitions of metadata on renaming?
  - Does `export_metadata`/`replace_metadata` supports the new metadata added?


#### GraphQL
   
   

#### Breaking changes


  1. Metadata API

     Existing `query` types:
     

  2. GraphQL API

     Schema Generation:
     
     
     

     Schema Resolve:-
     
     

  3. Logging
